cpu/fe310/clock.c

```diff
@@ -41,64 +39,67 @@ void clock_init(void)
 
-    /* Now it is safe to check for PLL Lock */
-    while ((PRCI_REG(PRCI_PLLCFG) & PLL_LOCK(1)) == 0);
-#endif
+            /* Now it is safe to check for PLL Lock */
+            while ((PRCI_REG(PRCI_PLLCFG) & PLL_LOCK(1)) == 0);
+        }
 
-    /* Switch over to PLL Clock source */
-    PRCI_REG(PRCI_PLLCFG) |= PLL_SEL(PLL_SEL_PLL);
+        /* Switch over to PLL Clock source */
+        PRCI_REG(PRCI_PLLCFG) |= PLL_SEL(PLL_SEL_PLL);
 
-    /* Turn off the HFROSC */
-    PRCI_REG(PRCI_HFROSCCFG) &= ~ROSC_EN(1);
-#elif CONFIG_USE_CLOCK_HFROSC_PLL
-    PRCI_set_hfrosctrim_for_f_cpu(CONFIG_CLOCK_DESIRED_FREQUENCY, PRCI_FREQ_UNDERSHOOT);
-#else /* Clock HFROSC */
-    /* Disable Bypass */
-    PRCI_REG(PRCI_PLLCFG) &= ~PLL_BYPASS(1);
+        /* Turn off the HFROSC */
+        PRCI_REG(PRCI_HFROSCCFG) &= ~ROSC_EN(1);
+    }
+    else if (CONFIG_USE_CLOCK_HFROSC_PLL) {
+        PRCI_set_hfrosctrim_for_f_cpu(CONFIG_CLOCK_DESIRED_FREQUENCY, PRCI_FREQ_UNDERSHOOT);
+    }
+    else { /* Clock HFROSC */
+        /* Disable Bypass */
+        PRCI_REG(PRCI_PLLCFG) &= ~PLL_BYPASS(1);
 
-    /* Configure trim and divider values of HFROSC */
-    PRCI_REG(PRCI_HFROSCCFG) = (ROSC_DIV(CONFIG_CLOCK_HFROSC_DIV) | ROSC_TRIM(CONFIG_CLOCK_HFROSC_TRIM) | ROSC_EN(1));
+        /* Configure trim and divider values of HFROSC */
+        PRCI_REG(PRCI_HFROSCCFG) = (ROSC_DIV(CONFIG_CLOCK_HFROSC_DIV) | ROSC_TRIM(CONFIG_CLOCK_HFROSC_TRIM) | ROSC_EN(1));
 
-    /* Wait for HFROSC to be ready */
-    while ((PRCI_REG(PRCI_HFROSCCFG) & ROSC_RDY(1)) == 0);
+        /* Wait for HFROSC to be ready */
+        while ((PRCI_REG(PRCI_HFROSCCFG) & ROSC_RDY(1)) == 0);
 
-    /* Don't use PLL clock source */
-    PRCI_REG(PRCI_PLLCFG) &= ~PLL_SEL(PLL_SEL_PLL);
-#endif
+        /* Don't use PLL clock source */
+        PRCI_REG(PRCI_PLLCFG) &= ~PLL_SEL(PLL_SEL_PLL);
+    }
 }
 
 uint32_t cpu_freq(void)
 {
-#if CONFIG_USE_CLOCK_HFXOSC || CONFIG_USE_CLOCK_HFXOSC_PLL
-    return CLOCK_CORECLOCK;
-#else /* Clock frequency with HFROSC cannot be determined precisely from
-         settings */
-    /* If not done already, estimate the CPU frequency */
-    if (_cpu_frequency == 0) {
-        /* Ignore the first run (for icache reasons) */
-        _cpu_frequency = PRCI_measure_mcycle_freq(3000, RTC_FREQ);
-        _cpu_frequency = PRCI_measure_mcycle_freq(3000, RTC_FREQ);
+    if (CONFIG_USE_CLOCK_HFROSC || CONFIG_USE_CLOCK_HFROSC_PLL) {
+        /* Clock frequency with HFROSC cannot be determined precisely from
+           settings */
+        /* If not done already, estimate the CPU frequency */
+        if (_cpu_frequency == 0) {
+            /* Ignore the first run (for icache reasons) */
+            _cpu_frequency = PRCI_measure_mcycle_freq(3000, RTC_FREQ);
+            _cpu_frequency = PRCI_measure_mcycle_freq(3000, RTC_FREQ);
+        }
+        return _cpu_frequency;
     }
-    return _cpu_frequency;
-#endif
+
+    return CLOCK_CORECLOCK;
```

Comment:
`CLOCK_CORECLOCK` seems to be undefined when compiling with ` CFLAGS="-DCONFIG_USE_CLOCK_HFROSC_PLL" `

Reply:
Ah, yes. I'm wondering how this should be fixed. Would it be ok to define CLOCK_CORECLOCK to 0 (or the desired frequency) when `CONFIG_USE_CLOCK_HFROSC_PLL` or `CONFIG_USE_CLOCK_HFROSC` clocks are used as system clock ? That would be a false define somehow.
The other fix is to keep the old version of this function.

Reply:
> That would be a false define somehow.

Even setting the desired frequency as you proposed would be wrong?

Reply:
The macro is not supposed to be accessed directly anyways, or is it?

Reply:
When using `CONFIG_USE_CLOCK_HFROSC_PLL` or `CONFIG_USE_CLOCK_HFROSC`, the macro CLOCK_CORECLOCK cannot be used because there's too much difference between the expected clock and the actual one. This is why there's some logic to estimate the current clock at runtime.

Reply:
I'll revert that change.

Reply:
Reverted